File: test/data/typescript/test05/StateState1.ts
import {BaseState} from "./BaseState"
import {IControllee} from "./IControllee"
import {StateController} from "./StateController"
export class StateState1 extends BaseState
{
    private _stateController: StateController;
    private _controlleeImp: IControllee;
    public constructor(stateController: StateController, controllee: IControllee)
    {
        super(controllee);
        this._stateController = stateController;
        this._controlleeImp = controllee;
    }
    public TransitEvent2(): BaseState | null
    {
        this._controlleeImp.DoAction2();
        this._stateController.InstanceOfState2.Setup(false, false);
        return this._stateController.InstanceOfState2;
    }
    public TransitEvent21(): BaseState | null
    {
        this._controlleeImp.DoAction21();
        this._stateController.InstanceOfState21.Setup(false, false);
        return this._stateController.InstanceOfState21;
    }
    public TransitEvent31(): BaseState | null
    {
        this._controlleeImp.DoAction31();
        this._stateController.InstanceOfState31.Setup(false, false);
        return this._stateController.InstanceOfState31;
    }
    public  GetStateName(): string
    {
        return "State1";
    }
    public GetParent(): BaseState | null
    {
        return null;
    }
    public GetStateID(): number
    {
        return 0;
    }
}

File: test/data/cs/test06/StateState1.cs
public class StateState1 : BaseState
{
    private StateController _stateController;
    private IControllee _controllee;
    public StateState1(StateController stateController, IControllee controllee) : base(controllee)
    {
        _stateController = stateController;
        _controllee = controllee;
    }
    public override BaseState? TransitAborted()
    {
        return null;
    }
    public override BaseState? TransitSucceeded()
    {
        _stateController.InstanceOfState2.Setup(false, false);
        return _stateController.InstanceOfState2;
    }
    public override string GetStateName()
    {
        return "State1";
    }
    public override BaseState? GetParent()
    {
        return null;
    }
    public override int GetStateID()
    {
        return 1;
    }
}